internal/services/shared/errors.go

```diff
@@ -104,6 +149,14 @@ func RewriteError(ctx context.Context, err error) error {
 	case errors.As(err, &relationNotFoundError):
 		return spiceerrors.WithCodeAndReason(err, codes.FailedPrecondition, v1.ErrorReason_ERROR_REASON_UNKNOWN_RELATION_OR_PERMISSION)
 
+	case errors.As(err, &maxDepthError):
+		if config == nil {
+			return spiceerrors.MustBugf("missing config for API error")
+		}
+
+		_, isCheckRequest := maxDepthError.Request.(*dispatchv1.DispatchCheckRequest)
+		return NewMaxDepthExceededError(config.MaximumAPIDepth, isCheckRequest)
```

Comment:
why not getting that config parameter from the dispatch error instead?

Reply:
Because the dispatch error doesn't have the maximum depth and we'd have to plumb it all the way through; dispatch simply takes in the *current* depth remaining, not the maximum depth overall